internal/sci/sci.proto

```diff
@@ -1,11 +1,25 @@
 syntax = "proto3";
 
 package sci.v1;
+import "google/protobuf/struct.proto";
 option go_package = "github.com/substratusai/substratus/internal/sci";
 
 service Controller {
   rpc CreateSignedURL(CreateSignedURLRequest) returns (CreateSignedURLResponse) {}
   rpc GetObjectMd5(GetObjectMd5Request) returns (GetObjectMd5Response) {}
+  rpc BindKSAToIAMPrincipal(BindKSAToIAMPrincipalRequest) returns (BindKSAToIAMPrincipalResponse) {} // strictly adds a permission for an IAM principal to a KSA
+}
+
+message BindKSAToIAMPrincipalRequest {
+  string namespace = 1;
+  string kubernetesServiceAccount = 2;
+  string principal = 3;
+  google.protobuf.Struct additionalParams = 4;
```

Comment:
We started with just `params` I thought maybe `policy` could work. This is somewhat flexible.

TO give a sense of what other cloud providers require:
* Using the `az` command line tool: https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html#federated-identity-credential-for-a-user-assigned-managed-identity
* Here's the go package and method in AWS: https://docs.aws.amazon.com/sdk-for-go/api/service/iam/#IAM.UpdateAssumeRolePolicy

Reply:
The issue is that some clouds will require additional information and others do not. However at a minimum all clouds will require namespace, KSA and IAM principal. Any additional details will have to be passed through params.

I think that would make it hard for the controller to know when to submit what information. So in hindsight. I think it makes sense to expand the gRPC call over time and for now add all info needed for AWS and GCP in the actual request.

Reply:
Updated to reflect what will work for GCP and AWS. AWS actually looks a lot like Azure in this domain so what I've got now should work for our immediate platforms.